code/edge/src/androidTest/java/com/adobe/marketing/mobile/RestartFunctionalTests.java

```diff
@@ -161,11 +156,7 @@ public void testRetryResponseQueuesEvents_restartSDK_verifyEventsDispatched_when
 	}
 
 	public void resetCore() throws Exception {
-		MobileCore.resetSDK();
-		MobileCore.setLogLevel(LoggingMode.VERBOSE);
-		Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
-		Application application = Instrumentation.newApplication(FunctionalTestHelper.CustomApplication.class, context);
-		MobileCore.setApplication(application);
+		FunctionalTestHelper.resetCoreHelper();
```

Comment:
This helper also calls resetServiceProvider which clears the db cache and some of the restart tests still assume there are some cached hits from "previous session". Can we make the clear db optional for those tests?

Reply:
It calls `ServiceProvider.resetServiceProvider` which just reinitializes the services. It does not call `FunctionalTestHelper.resetServiceProvider` which additionally clears the shared preferences, application, and database directories.